utils.go

```diff
@@ -122,22 +126,26 @@ func (s *Storage) formatError(op string, err error, path ...string) error {
 	}
 }
 
-func (s *Storage) formatFileObject(v *upyun.FileInfo) (o *types.Object, err error) {
-	o = &types.Object{
-		ID:         v.Name,
-		Name:       s.getRelPath(v.Name),
-		Type:       types.ObjectTypeFile,
-		Size:       v.Size,
-		UpdatedAt:  v.Time,
-		ObjectMeta: info.NewObjectMeta(),
-	}
+func (s *Storage) formatFileObject(v *upyun.FileInfo) (o *typ.Object, err error) {
+	o = s.newObject(false)
+	o.ID = v.Name
+	o.Path = s.getRelPath(v.Name)
+	o.Mode |= typ.ModeRead
+
+	o.SetContentLength(v.Size)
+	o.SetLastModified(v.Time)
+	o.SetServiceMetadata(v.Meta)
 
-	if v.ETag != "" {
-		o.SetETag(v.ETag)
+	if v.MD5 != "" {
+		o.SetEtag(v.MD5)
```

Comment:
`ETag` is different from `MD5`.

If service provider makes it sure that they returns a ContentMd5, we need to use `SetContentMd5`.
If service provider makes it clear that they use object's md5 value to differ object's content, we can use `SetEtag`.